mantidimaging/core/operations/median_filter/median_filter.py

```diff
@@ -83,19 +82,32 @@ def filter_func(data: ImageStack, size=None, mode="reflect", progress=None, forc
         :return: Returns the processed data
 
         """
+        # Validation
         h.check_data_stack(data)
-
-        if not size or not size > 1:
+        if size is None or size <= 1:
             raise ValueError(f'Size parameter must be greater than 1, but value provided was {size}')
 
-        if not force_cpu:
-            _execute_gpu(data.data, size, mode, progress)
-        else:
-            _execute(data, size, mode, progress)
+        params = {'mode': mode, 'force_cpu': force_cpu}
+        if force_cpu:
+            params['size'] = size  # Pass size only if using CPU
+        ps.run_compute_func(MedianFilter.compute_function, data.data.shape[0], data.shared_array, params)
 
         h.check_data_stack(data)
         return data
 
+    @staticmethod
+    def compute_function(i: int, array: np.ndarray, params: Dict[str, Any]):
+        mode = params['mode']
+        force_cpu = params['force_cpu']
+        size = params.get('size')
+        progress = params.get('progress')
+
+        if not force_cpu:
+            cuda = gpu.CudaExecuter(array.dtype)
+            cuda.median_filter(i, array, mode=mode, progress=progress)  # Call without size if it is None
+        else:
+            array[i] = _median_filter(array[i], size=size if size is not None else 3, mode=mode)
```

Comment:
It would be good to keep all of this logic outside the compute function. Its only the CPU path that needs to use the compute function, the cuda one can stay unchanged. (as it is now, it will start lots of cuda jobs which is not what we want).